src/gennylib/src/encryption.cpp

```diff
@@ -453,16 +453,42 @@ class EncryptionManager::EncryptionManagerImpl {
 
     // whether setup that needs a server connection should be skipped
     bool _dryRun;
+
+    // whether to use shared library instead of mongocryptd
+    bool _useCryptSharedLib = false;
+
+    // path to the mongo crypt shared library
+    std::string _cryptSharedLibPath;
 };
 
 EncryptionManager::EncryptionManagerImpl::EncryptionManagerImpl(const Node& yaml, bool dryRun)
     : _dryRun(dryRun) {
-    const auto& collsSequence = yaml["EncryptedCollections"];
+    const auto& encryptionNode = yaml["Encryption"];
+    if (!encryptionNode) {
+        return;
+    }
+
+    auto useShlibOpt = encryptionNode["UseCryptSharedLib"].maybe<bool>();
+    if (useShlibOpt) {
+        _useCryptSharedLib = *useShlibOpt;
+    }
+    if (_useCryptSharedLib) {
+        auto shlibPathOpt = encryptionNode["CryptSharedLibPath"].maybe<std::string>();
+        if (!shlibPathOpt || shlibPathOpt->empty()) {
```

Comment:
Might wanna do `boost::filesystem::exists` as a basic validation too

Reply:
libmongocrypt already throws a reasonable error message when it fails to `dlopen` this file name on client pool creation, so I opted not to do a redundant check here.